iOS/CodePushUtils.m

```diff
@@ -0,0 +1,156 @@
+        NSString * fullFilePath = [sourceFolder stringByAppendingPathComponent:fileName];
+        BOOL isDir = NO;
+        if ([[NSFileManager defaultManager] fileExistsAtPath:fullFilePath
+                                                 isDirectory:&isDir] && isDir) {
+            NSString *nestedDestFolder = [destFolder stringByAppendingPathComponent:fileName];
+            [self copyEntriesInFolder:fullFilePath
+                           destFolder:nestedDestFolder
+                                error:error];
+        } else {
+            NSString *destFileName = [destFolder stringByAppendingPathComponent:fileName];
+            if ([[NSFileManager defaultManager] fileExistsAtPath:destFileName]) {
+                [[NSFileManager defaultManager] removeItemAtPath:destFileName error:error];
+                if (*error) {
+                    return;
+                }
+            }
+            if (![[NSFileManager defaultManager] fileExistsAtPath:destFolder]) {
+                [[NSFileManager defaultManager] createDirectoryAtPath:destFolder
+                                          withIntermediateDirectories:YES
+                                                           attributes:nil
+                                                                error:error];
+                if (*error) {
+                    return;
+                }
+            }
+            
+            [[NSFileManager defaultManager] copyItemAtPath:fullFilePath toPath:destFileName error:error];
+            if (*error) {
+                return;
+            }
+        }
+    }
+}
+
++ (NSString *)findMainBundleInFolder:(NSString *)folderPath
+                               error:(NSError **)error
+{
+    NSArray* folderFiles = [[NSFileManager defaultManager]
+                            contentsOfDirectoryAtPath:folderPath
+                            error:error];
+    if (*error) {
+        return nil;
+    }
+    
+    for (NSString *fileName in folderFiles) {
+        NSString *fullFilePath = [folderPath stringByAppendingPathComponent:fileName];
+        BOOL isDir = NO;
+        if ([[NSFileManager defaultManager] fileExistsAtPath:fullFilePath
+                                                 isDirectory:&isDir] && isDir) {
+            NSString *mainBundlePathInFolder = [self findMainBundleInFolder:fullFilePath error:error];
+            if (*error) {
+                return nil;
+            }
+            
+            if (mainBundlePathInFolder) {
+                return [fileName stringByAppendingPathComponent:mainBundlePathInFolder];
+            }
+        } else if ([[fileName pathExtension] isEqualToString:@"bundle"] ||
+                   [[fileName pathExtension] isEqualToString:@"jsbundle"] ||
+                   [[fileName pathExtension] isEqualToString:@"js"]) {
+            return fileName;
+        }
+    }
+    
+    return nil;
+}
+
++ (BOOL)verifyHashForZipUpdate:(NSString *)finalUpdateFolder
```

Comment:
So we dont want to compute the hash if it is a zip update but not a diff update? I guess i'm fairly confident the unzipping process won't go wrong.